packages/jss/src/rules/KeyframesRule.js

```diff
@@ -20,6 +28,10 @@ export default class KeyframesRule implements ContainerRule {
 
   constructor(key: string, frames: Object, options: RuleOptions) {
     this.key = key
+    // TODO make it more robust
+    this.name = key.substr(this.type.length + 2)
+    const fakeRule: Rule = ({key: this.name}: any)
+    this.id = options.jss.generateClassName(fakeRule, options.sheet)
```

Comment:
this is not very nice, I actually want id generation to be outside of a rule and passed to it